lib/src/model/coursetable/course.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:flutter_app/src/model/coursetable/course_period.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+import '../json_init.dart';
+
+part 'course.g.dart';
+
+@JsonSerializable()
+class Course {
+  String idString;
+  late int id;
+
+  String name;
+
+  String stageString;
+  late double stage;
+
+  String creditString;
+  late double credit;
+
+  String periodCountString;
+  late int periodCount;
+
+  String category;
+
+  String teacherString;
+  late List<String> teachers;
+
+  String classNameString;
+  late List<String> classNames;
+
+  List<String> periodSlots;
+  late List<CoursePeriod> coursePeriods;
+
+  String classroomString;
+  late List<String> classrooms;
+
+  String applyStatus;
+  String language;
+  String syllabusLink;
+  String note;
+
+  Course(
+      {required this.idString,
+      required this.name,
+      required this.stageString,
+      required this.creditString,
+      required this.periodCountString,
+      required this.category,
+      required this.teacherString,
+      required this.classNameString,
+      required this.periodSlots,
+      required this.classroomString,
+      required this.applyStatus,
+      required this.language,
+      required this.syllabusLink,
+      required this.note}) {
+    id = JsonInit.intInit(idString);
+    name = JsonInit.stringInit(name).trim();
+    stage = JsonInit.doubleInit(stageString);
+    credit = JsonInit.doubleInit(creditString);
+    periodCount = JsonInit.intInit(periodCountString);
+    category = JsonInit.stringInit(category).trim();
+    teacherString = JsonInit.stringInit(teacherString).trim();
+    teachers = teacherString.split(RegExp(r"\n")).map((element) => element.trim()).toList();
+    periodSlots = JsonInit.listInit<String>(periodSlots);
+    coursePeriods = _convertPeriodSlotsToCoursePeriods(periodSlots);
+    classroomString = JsonInit.stringInit(classroomString).trim();
+    classrooms = classroomString.split(RegExp(r"\n")).map((element) => element.trim()).toList();
+    classNameString = JsonInit.stringInit(classNameString).trim();
+    classNames = classNameString.split(RegExp(r"\n")).map((element) => element.trim()).toList();
+    applyStatus = JsonInit.stringInit(applyStatus).trim();
+    language = JsonInit.stringInit(language).trim();
+    syllabusLink = JsonInit.stringInit(syllabusLink).trim();
+    note = JsonInit.stringInit(note).trim();
```

Comment:
I'd suggest not doing huge data content checks in the constructor, because:

1. Maybe cause unneeded performance waste.
Although we expect some of data may include strange things, we can still do such check when we really need to use them (at other layer of architecture, separating Model and DTO). That means, if we only need variable `a`, then we just check `a` without checking `b`, if `a` is not based on `b`.

2. `JsonSerializable` will actually help us do most of things like this. 
In the generated `.g.dart` file, there will be several general solutions of string parsing be implemented, if we have clearly declared variable data type, even it is a nullable type.

Reply:
As above: https://github.com/NEO-TAT/tat_flutter/pull/221#discussion_r1485888375

Reply:
Suggest following https://github.com/NEO-TAT/tat_flutter/pull/221#issuecomment-1940438630.